scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -130,69 +109,39 @@ def create_ngram_index(
                     for reference_ngram in ngrams(reference_unigrams, n):
                         if reference_ngram not in ngram_index[n]:
                             ngram_index[n][reference_ngram] = set()
+                        id = instance.id if instance.id else f"id{i}"
                         ngram_index[n][reference_ngram].add(
-                            EntryDataOverlapKey(stats_key=stats_key, instance_id=i, part=PART_REF)
+                            EntryDataOverlapKey(stats_key=stats_key, instance_id=id, index=i, part=PART_REF)
                         )
     return ngram_index
 
 
-def create_all_data_overlap_stats(light_scenarios: List[LightScenario], n_values: List[int]) -> AllDataOverlapStats:
-    """Given a list of scenarios and n values, initialize all_overlap_stats"""
-    hlog("Initializing all data overlap stats")
-    all_overlap_stats: AllDataOverlapStats = {}
-    for scenario in light_scenarios:
-        for n in n_values:
-            # Initialize a stats instance for every pair of <scenario, n>
-            stats: DataOverlapStats = DataOverlapStats.from_scenario(scenario, stats_tags={"N": n})
-            if stats.stats_key in all_overlap_stats:
-                raise ValueError("Duplicated settings detected.")
-            all_overlap_stats[stats.stats_key] = stats
-    return all_overlap_stats
-
-
-def compute_scenario_file_data_overlap(
+def compute_all_data_overlap(
     training_file_path: str,
     file_format: str,
     ngram_index: NgramIndex,
-    all_overlap_stats: AllDataOverlapStats,
     tokenizer: LightTokenizer,
-    ngram_counter: Optional[NgramCounter] = None,
-    max_overlapping_ngrams: int = 0,
+    stats_key_to_input_ids: Dict[DataOverlapStatsKey, Set[str]],
+    stats_key_to_reference_ids: Dict[DataOverlapStatsKey, Set[str]],
```

Comment:
Optional: `stats_key_to_overlapping_instance_ids` and `stats_key_to_overlapping_reference_ids`

Reply:
I'll probably keep it as is since we we use 'input' and 'reference' as fields, and overloading instance can be confusing